### DISCUSSION TITLE
Crazyflie flies spiral and crashes when giving simple takeoff command.

### DISCUSSION THREAD

**Discussion by tangwh20** [bitcraze/discussions#1031] [Q&A]
Hi, everyone. I'm using **Crazyflie with Optitrack mocap system**. I followed all the instructions posted on the official website, and set up the **crazyswarm2** environment introduced in https://imrclab.github.io/crazyswarm2/ . But my Crazyflie almost always fails to take off smoothly (almost always means once in a while it accidentally succeeds, and after some times it fails again). I simply run a **hello-world script**, where it should simply **take off and land**. But in most cases it follows a weird **spiral trajectory**, where it circles and crashes to the boundary of the field. All firmwares are all right and there are no errors thrown out for the whole period. One of the videos is posted below for reference. 

https://github.com/bitcraze/discussions/assets/99193853/8d2ecb51-270d-472d-ae50-0079bb8785f5

We have tried methods including **replacing motors and propellers, switching to crazyswarm1 environment, changing mocap cameras and experiment sites**. But none of these work for our problem.

Can anyone explain why this happens? Thanks a lot!

**Comment by knmcguire** [comment]
Hi! It seems that things are going wrong with the positioning as this behavior is usually observed when the crazyflie does not have a good position estimate.

Crazyswarm2 specific questions should be asked in https://github.com/IMRCLab/crazyswarm2/discussions. You could however, try out the [mocap example in our crazyflie library](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/mocap/mocap_hl_commander.py). That is perhaps a simpler example and it uses the same mocap library made for crazyswarm2, so if that doesn't fly then you know better where the issue is.
